src/tests/integration/omneo/rewards/delete-reward.test.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, expect, test, afterAll } from 'vitest'
+import { Omneo } from '../../../../omneo'
+import simpleOmneoRequest from '../../../lib/simple-omneo-request'
+import { RewardDefinitionCreateInput } from '../../../../types'
+import { getName, getHandle, getExternalId } from './util'
+
+const omneo = new Omneo({
+  tenant: process.env.OMNEO_TENANT as string,
+  token: process.env.OMNEO_TOKEN as string
+})
+const CREATED_REWARDS_DEFINITION_IDS : number[] = []
+const FAILED_REWARDS_IDS : number[] = []
+const testProfileID = process.env.OMNEO_TEST_PROFILE_ID as string
+
+describe('Reward delete', () => {
+  test('SDK Reward delete', async () => {
+    const payload: RewardDefinitionCreateInput = {
+      name: getName(),
+      handle: getHandle(),
+      value: 10,
+      period: 30,
+      period_type: 'days',
+      type: 'spend',
+      is_assignable: true,
+      is_published: true
+    }
+    const response = await simpleOmneoRequest('POST', '/rewards/definitions', payload).catch((err) => {
+      console.error('SDK update reward definition created failed:', err)
+      throw new Error('SDK update reward definition created failed')
+    })
+    CREATED_REWARDS_DEFINITION_IDS.push(response.data.id)
+
+    const rewardPayload = {
+      reward_definition_id: response.data.id,
+      profile_id: testProfileID,
+      value_initial: 5,
+      value_remaining: 5,
+      expires_at: '2024-12-06 08:30:00',
+      issued_at: '2024-12-05 08:30:00',
+      timezone: 'Australia/Melbourne',
+      external_id: getExternalId()
+    }
+    const rewardResponse = await simpleOmneoRequest('POST', '/rewards', rewardPayload).catch((err) => {
+      console.error('SDK list rewards created failed:', err)
+      throw new Error('SDK list rewards created failed')
+    })
+
+    const rewardsRes = await omneo.rewards.list({
```

Comment:
fetch by ID instead and expect a 404 response

Reply:
This is list by external id, so not return 404 response.

Reply:
use simpleOmneoRequest to get by ID
The id is in your `rewardResponse`

You shouldn't use the SDK, and also shouldn't use the list endpoint. 
You should get your specific reward you created